cpu/cc2538/Makefile.cc2538

```diff
@@ -84,7 +85,7 @@ CUSTOM_RULE_LINK=1
 
 %.elf: $(CPU_STARTFILES) %.co $(PROJECT_OBJECTFILES) $(PROJECT_LIBRARIES) contiki-$(TARGET).a $(LDSCRIPT)
 	$(TRACE_LD)
-	$(Q)$(LD) $(LDFLAGS) ${filter-out $(LDSCRIPT) %.a,$^} ${filter %.a,$^} $(TARGET_LIBFILES) -o $@
+	$(Q)$(LD) $(LDFLAGS) -Wl,--start-group ${filter-out $(LDSCRIPT) %.a,$^} ${filter %.a,$^} $(TARGET_LIBFILES) -lc -Wl,--end-group -o $@
```

Comment:
Thanks for taking the time to share your thoughts and for the in-depth analysis.

I'll shit this from a line note to the PR's discussion, cause a `pull -f` will hide it.